From 116f2588af62433512e822525f2804d3288b9252 Mon Sep 17 00:00:00 2001
From: Toshifumi NISHINAGA <tnishinaga.dev@gmail.com>
Date: Sat, 6 Jul 2019 04:25:10 +0000
Subject: [PATCH 1/3] Add dwc_otg usb driver from plathome kernel

original is here:
ftp://ftp.plathome.co.jp/pub/OBS600/unofficial/jessie/4.1.8-0/source/linux-4.1.8-obs-20151016.tar.xz
---
 arch/powerpc/platforms/40x/Makefile     |    1 +
 arch/powerpc/platforms/40x/obs600-usb.c |  116 +
 drivers/usb/Kconfig                     |    2 +
 drivers/usb/Makefile                    |    2 +
 drivers/usb/dwc_otg/Kconfig             |   43 +
 drivers/usb/dwc_otg/Makefile            |   38 +
 drivers/usb/dwc_otg/dwc_otg_attr.c      |  802 ++++++
 drivers/usb/dwc_otg/dwc_otg_attr.h      |   67 +
 drivers/usb/dwc_otg/dwc_otg_cil.c       | 3029 +++++++++++++++++++++++
 drivers/usb/dwc_otg/dwc_otg_cil.h       |  911 +++++++
 drivers/usb/dwc_otg/dwc_otg_cil_intr.c  |  708 ++++++
 drivers/usb/dwc_otg/dwc_otg_driver.c    | 1244 ++++++++++
 drivers/usb/dwc_otg/dwc_otg_driver.h    |   84 +
 drivers/usb/dwc_otg/dwc_otg_hcd.c       | 2867 +++++++++++++++++++++
 drivers/usb/dwc_otg/dwc_otg_hcd.h       |  676 +++++
 drivers/usb/dwc_otg/dwc_otg_hcd_intr.c  | 1842 ++++++++++++++
 drivers/usb/dwc_otg/dwc_otg_hcd_queue.c |  794 ++++++
 drivers/usb/dwc_otg/dwc_otg_plat.h      |  303 +++
 drivers/usb/dwc_otg/dwc_otg_regs.h      | 1797 ++++++++++++++
 19 files changed, 15325 insertions(+)
 create mode 100644 arch/powerpc/platforms/40x/obs600-usb.c
 create mode 100644 drivers/usb/dwc_otg/Kconfig
 create mode 100644 drivers/usb/dwc_otg/Makefile
 create mode 100644 drivers/usb/dwc_otg/dwc_otg_attr.c
 create mode 100644 drivers/usb/dwc_otg/dwc_otg_attr.h
 create mode 100644 drivers/usb/dwc_otg/dwc_otg_cil.c
 create mode 100644 drivers/usb/dwc_otg/dwc_otg_cil.h
 create mode 100644 drivers/usb/dwc_otg/dwc_otg_cil_intr.c
 create mode 100644 drivers/usb/dwc_otg/dwc_otg_driver.c
 create mode 100644 drivers/usb/dwc_otg/dwc_otg_driver.h
 create mode 100644 drivers/usb/dwc_otg/dwc_otg_hcd.c
 create mode 100644 drivers/usb/dwc_otg/dwc_otg_hcd.h
 create mode 100644 drivers/usb/dwc_otg/dwc_otg_hcd_intr.c
 create mode 100644 drivers/usb/dwc_otg/dwc_otg_hcd_queue.c
 create mode 100644 drivers/usb/dwc_otg/dwc_otg_plat.h
 create mode 100644 drivers/usb/dwc_otg/dwc_otg_regs.h

diff --git a/arch/powerpc/platforms/40x/Makefile b/arch/powerpc/platforms/40x/Makefile
index 828d78340dd9..c8078b74eba0 100644
--- a/arch/powerpc/platforms/40x/Makefile
+++ b/arch/powerpc/platforms/40x/Makefile
@@ -1,4 +1,5 @@
 # SPDX-License-Identifier: GPL-2.0-only
+#obj-$(CONFIG_OBS600)				+= obs600-usb.o
 obj-$(CONFIG_WALNUT)				+= walnut.o
 obj-$(CONFIG_XILINX_VIRTEX_GENERIC_BOARD)	+= virtex.o
 obj-$(CONFIG_EP405)				+= ep405.o
diff --git a/drivers/usb/Kconfig b/drivers/usb/Kconfig
index 275568abc670..6a48e1545f5e 100644
--- a/drivers/usb/Kconfig
+++ b/drivers/usb/Kconfig
@@ -117,6 +117,8 @@ source "drivers/usb/mtu3/Kconfig"
 
 source "drivers/usb/musb/Kconfig"
 
+source "drivers/usb/dwc_otg/Kconfig"
+
 source "drivers/usb/dwc3/Kconfig"
 
 source "drivers/usb/dwc2/Kconfig"
diff --git a/drivers/usb/Makefile b/drivers/usb/Makefile
index 1c1c1d659394..beeefd401214 100644
--- a/drivers/usb/Makefile
+++ b/drivers/usb/Makefile
@@ -37,6 +37,8 @@ obj-$(CONFIG_USB_MAX3421_HCD)	+= host/
 
 obj-$(CONFIG_USB_C67X00_HCD)	+= c67x00/
 
+obj-$(CONFIG_DWC_OTG)		+= dwc_otg/
+
 obj-$(CONFIG_USB_ACM)		+= class/
 obj-$(CONFIG_USB_PRINTER)	+= class/
 obj-$(CONFIG_USB_WDM)		+= class/
-- 
2.26.0

